### PR TITLE
Fix for [Sticky] failes to init and remove html markup #8156

### PR DESCRIPTION
**Pull request by randmanrjr**
I am submitting this pull request for the following issue: [https://github.com/zurb/foundation-sites/issues/8156](https://github.com/zurb/foundation-sites/issues/8156)

### CODE REVIEW

File: js/foundation.sticky.js [hunk 1]
```diff
@@ -38,7 +38,13 @@ class Sticky {
...
       this.wasWrapped = true;
     }
-    this.$container = $parent.length ? $parent : $(this.options.container).wrapInner(this.$element);
+      if($parent.lenth){
+          this.$container = $parent;
+      } else {
+          this.wasWrapped = true;
+          this.$element.wrap(this.options.container);
+          this.$container = this.$element.parent();
+      }
```
Comment (IamManchanda) [inline js/foundation.sticky.js]: Proper indentation please ??

File: js/foundation.sticky.js [hunk 2]
```diff
@@ -38,7 +38,13 @@ class Sticky {
     if (!$parent.length) {
       this.wasWrapped = true;
     }
-    this.$container = $parent.length ? $parent : $(this.options.container).wrapInner(this.$element);
+      if($parent.lenth){
+          this.$container = $parent;
+      } else {
```
Comment (IamManchanda) [inline js/foundation.sticky.js]: @randmanrjr Proper indentation please ??

**Review by IamManchanda — CHANGES_REQUESTED** [review]

File: js/foundation.sticky.js [hunk 3]
```diff
@@ -38,7 +38,13 @@ class Sticky {
     if (!$parent.length) {
       this.wasWrapped = true;
     }
-    this.$container = $parent.length ? $parent : $(this.options.container).wrapInner(this.$element);
+    if($parent.lenth){
+        this.$container = $parent;
```
Comment (IamManchanda) [inline js/foundation.sticky.js]: @randmanrjr Still need a bit more indentation fix 
=>  If confused, compare line 39 and 42 !

{{ Yes i am annoying :smile: }}

Comment (randmanrjr) [inline js/foundation.sticky.js]: No problem. Code style and quality control are important :-)

File: js/foundation.sticky.js [hunk 4]
```diff
@@ -38,7 +38,13 @@ class Sticky {
     if (!$parent.length) {
       this.wasWrapped = true;
     }
-    this.$container = $parent.length ? $parent : $(this.options.container).wrapInner(this.$element);
+    if($parent.lenth){
```
Comment (IllusionPerdu) [inline js/foundation.sticky.js]: Why you don't use the `if (!$parent.length) {}` in line 38 ?

File: js/foundation.sticky.js [hunk 5]
```diff
@@ -35,10 +35,13 @@ class Sticky {
         id = this.$element[0].id || Foundation.GetYoDigits(6, 'sticky'),
         _this = this;
 
-    if (!$parent.length) {
+    if($parent.lenth){
```
Comment (kball) [inline js/foundation.sticky.js]: typo - length

**Comment by kball** [comment]
Merged!  Thanks @randmanrjr! Congrats on first PR merged!

**Comment by randmanrjr** [comment]
@kball, You're most welcome! I use Foundation every day, so it's nice to be able to give back some love.

**Pull request merged**